android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -450,6 +452,21 @@ constructor(
       }
       return serviceStatus
     }
+
+    fun generateListTaskServiceStatus(tasks: List<Task>): String {
+      val anyOverDueTask = tasks.find { task -> task.isOverDue() }
+      val anyDueTask = tasks.find { task -> task.isDue() }
+      val anyUpcomingTask = tasks.find { task -> task.isUpcoming() }
+      val anyInProgressTask = tasks.find { task -> task.status == Task.TaskStatus.INPROGRESS }
+      val anyExpiredTask = tasks.find { task -> task.status == Task.TaskStatus.CANCELLED }
+      val anyCompletedTask = tasks.find { task -> task.status == Task.TaskStatus.COMPLETED }
```

Comment:
@AbdulWahabMemon I might be missing something. Why are we doing this here? My assumption is We want to return service status for different tasks. Why don't we just run a loop for the List of Tasks?

Reply:
Let's discuss this and make the necessary changes on the PR on `main`

Reply:
above method updated, now the logic will be executed with possible less iteration against entire task list loop iteration